modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/FrontendDataSet.js

```diff
@@ -273,6 +277,33 @@ const FrontendDataSet = ({
 		}
 	}
 
+	const pageSelectedItemsValue = selectedItemsValue.filter((id) =>
+		items.some((item) => item.id === id)
+	);
+
+	if (
+		allItemsSelectedActive &&
+		!!pageSelectedItemsValue.length &&
+		pageSelectedItemsValue.length !== items.length
+	) {
+		setAllItemsSelectedActive(false);
+		deselectItems(
+			selectedItemsValue.filter(
+				(item) => !pageSelectedItemsValue.includes(item)
+			)
+		);
+	}
+	else if (allItemsSelectedActive) {
```

Comment:
Minor point here, this can be simplified to 
```js
if (allItemsSelectedActive) {
    if (!!pageSelectedItemsValue.length && ... ) {
    } 
    else {
    }
}
```